improver/generate_ancillaries/generate_timezone_mask.py

```diff
@@ -148,8 +148,8 @@ def _get_coordinate_pairs(cube):
             longitudes = cube.coord("longitude").points.copy()
 
             # timezone finder works using -180 to 180 longitudes.
-            if (longitudes > 180).any():
-                longitudes[longitudes > 180] -= 180
+            if (longitudes > 180).any() and (longitudes >= 0).all():
```

Comment:
I notice that you've tried updating the environment to install `pytz`. As this is bit similar to what I saw in #1394 when adding `statsmodels` to the environment, I think that you need to increment the [CACHE NUMBER](https://github.com/metoppv/improver/blob/master/.github/workflows/tests.yml#L17). Also, the first time the tests are run, it will create a new cache of the environment. The `py36_iris22` environment seems to take a long time to finish building (~2 hours!) but once the cache is built, subsequent runs will be quick (I've tested this in https://github.com/metoppv/improver/pull/1394/commits/75e848005ecb12669a9cfddfc02f1eb52d2a8dc6). Therefore, you could still consider adding `pytz` to the GitHub Actions environment, so that these unit tests are tested, as I don't think it's been tested how longer running the tests on GitHub Actions with `pytz` will take.